### ISSUE TITLE
Working distributed database in browser?

### ISSUE THREAD

**Issue by runvnc** [kgryte/awesome-peer-to-peer#5]
TLDR: which of the many projects listed here will actually work for wiki or database running entirely in-browser? (If any).

Hello, thanks so much for putting this together.  If you have a couple of minutes (I am sure you are busy so no rush getting to this), I could use some help.  I looked at a number of the projects.  The libp2p ones look interesting to me because the IPFS js project seems to be using it.  I also looked at a few other of the listed projects.

I am unable to determine if there is a stand-out or necessarily working distributed peer-to-peer database system or which one would be most capable or easiest to use.

My application is basically a wiki/database: https://www.reddit.com/r/Permaculture/comments/4vbmyr/where_can_i_find_existing_andor_tested_full/ Basically I intend to store three types of entities: Permaculture Full Designs, Permaculture Guilds, and Permaculture Plants.  Guilds are collections of plants, and designs are (mostly) collections of guilds.  I would need to store or reference images, metadata (JSON), and descriptive text or markup for all of these things.

Really any system that can support either a database or a wiki will do, but I want people to be able to run it in their browser, and have all of the data stored in the network of browser windows with no server (or maybe a headless client on a server serving as a backup node).

Thanks very much.

**Comment by kgryte** [comment]
@runvnc I would look at hyperdrive (and hyperlog) and the associated Node.js packages written and maintained by Mathias Buus (mafintosh), Max Odgen (maxodgen), and James Halliday (substack). They have done quite a bit of work around a p2p database system which can work (for the most part) entirely in-browser. See also the related `dat` project.
